x/foundation/keeper/keys.go

```diff
@@ -70,39 +79,38 @@ func voteKey(proposalID uint64, voter sdk.AccAddress) []byte {
 	return key
 }
 
-func proposalByVPEndKey(id uint64, end time.Time) []byte {
+func proposalByVPEndKey(vpEnd time.Time, id uint64) []byte {
+	prefix := proposalByVPEndKeyPrefix
+	vpEndBz := sdk.FormatTimeBytes(vpEnd)
 	idBz := Uint64ToBytes(id)
-	endBz := sdk.FormatTimeBytes(end)
-	key := make([]byte, len(proposalByVPEndKeyPrefix)+1+len(idBz)+len(endBz))
+	key := make([]byte, len(prefix)+lenTime+len(idBz))
 
 	begin := 0
-	copy(key[begin:], proposalByVPEndKeyPrefix)
+	copy(key[begin:], prefix)
 
-	begin += len(proposalByVPEndKeyPrefix)
-	key[begin] = byte(len(idBz))
+	begin += len(prefix)
+	copy(key[begin:], vpEndBz)
 
-	begin++
+	begin += len(vpEndBz)
 	copy(key[begin:], idBz)
 
-	begin += len(idBz)
-	copy(key[begin:], endBz)
-
 	return key
 }
 
-// func splitProposalByVPEndKey(key []byte) (proposalID uint64, vpEnd time.Time) {
-// 	begin := len(proposalByVPEndKeyPrefix) + 1
-// 	end := begin + int(key[begin-1]) // uint64
-// 	proposalID = Uint64FromBytes(key[begin:end])
+func splitProposalByVPEndKey(key []byte) (vpEnd time.Time, id uint64) {
+	prefix := proposalByVPEndKeyPrefix
+	begin := len(prefix)
+	end := begin + lenTime
+	vpEnd, err := sdk.ParseTimeBytes(key[begin:end])
+	if err != nil {
+		panic(err)
```

Comment:
Isn't there this panic case?
There is not unittest.

Reply:
You can add a unit test that covers this line. But in the sane condition, it won't happen. There are many similar lines, and of course, x/foundation is not only the case. I think it won't be necessary to add the unit test.

Reply:
OK, I understand.